src/main/kotlin/app/api/SourcererApi.kt

```diff
@@ -0,0 +1,39 @@
+// Copyright 2017 Sourcerer Inc. All Rights Reserved.
+// Author: Anatoly Kislov (anatoly@sourcerer.io)
+
+package app.api
+
+import app.model.Commit
+import app.model.Repo
+import app.model.User
+
+/**
+ * Sourcerer API.
+ */
+object SourcererApi : Api {
+    var api: Api = ServerApi()
```

Comment:
Your SourcererAPI is essentially a factory. As far as a consumer is concerned, all you use it for is to produce an instance of API that varies depending on purpose (production/testing). To be fair, even a factor is not needed here. A traditional way to address the need for tests is to pass an API to its user constructor on initialization:

```
var api = new ServerAPI();
var user = new SomeAPIUser(api);
```

...and off you go. Very rarely in engineering there is a need for singletons, and it's usually for something pervasive like logging.